### PR TITLE
Add OpenProcess and SafeHandle utilities

### PR DESCRIPTION
**Pull request by pierricgimmig**
"SafeHandle" is a wrapper around a Windows "HANDLE" which calls
"CloseHandle" on destruction if said handle is not null.

OpenProcess.h contains wrapper functions around Windows' "OpenProcess"
function which return a "SafeHandle" or an error.

Add SafeHandleTest.cpp, OpenProcessTest.cpp

Tests: Ran tests, compiled.

### CODE REVIEW

File: src/WindowsUtils/include/WindowsUtils/OpenProcess.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#include "WindowsUtils/SafeHandle.h"
+
+namespace orbit_windows_utils {
+
+// Wrapper around Windows' "OpenProcess" function which either returns a "SafeHandle" or an error.
+ErrorMessageOr<SafeHandle> OpenProcess(uint32_t desired_access, bool inherit_handle,
+                                       uint32_t process_id);
+// Calls OpenProcess(PROCESS_VM_READ, /*inherit_handle=*/false, process_id).
+ErrorMessageOr<SafeHandle> OpenProcess(uint32_t process_id);
```
Comment (danielfenner) [inline src/WindowsUtils/include/WindowsUtils/OpenProcess.h:16]: nit: I'd call this function `OpenProcessForReading` or similar.

File: src/WindowsUtils/SafeHandleTest.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "OrbitBase/ThreadUtils.h"
+#include "WindowsUtils/OpenProcess.h"
+#include "WindowsUtils/SafeHandle.h"
+
+using orbit_windows_utils::SafeHandle;
+
+TEST(SafeHandle, OwnershipTransfer) {
+  auto result = orbit_windows_utils::OpenProcess(orbit_base::GetCurrentProcessId());
+  EXPECT_TRUE(result.has_value());
```
Comment (danielfenner) [inline src/WindowsUtils/SafeHandleTest.cpp:16]: `ASSERT_TRUE(result.has_value());`

The access to `result.value()` below will fail - so we should not continue here.

**Review by danielfenner — APPROVED** [review]
lgtm - merely nits.

File: src/WindowsUtils/include/WindowsUtils/OpenProcess.h (hunk 1 above)
Comment (pierricgimmig) [reply src/WindowsUtils/include/WindowsUtils/OpenProcess.h]: Done, thanks.

File: src/WindowsUtils/SafeHandleTest.cpp (hunk 2 above)
Comment (pierricgimmig) [reply src/WindowsUtils/SafeHandleTest.cpp]: Done, thanks.

**Pull request merged**